server/src/Application/Services/Course/CourseService.cs

```diff
@@ -15,17 +16,17 @@ public CourseService(ICourseRepository courses, ILectureRepository lectures)
         _lectures = lectures;
     }
 
-    public Course CreateCourse(CreateCourseDto dto)
+    public CourseDto CreateCourse(CreateCourseDto dto)
     {
         var newCourse = new Course(name: dto.Name, description: dto.Description);
         _courses.Add(newCourse);
-        return newCourse;
+        return new CourseDto(newCourse);
     }
 
-    public Course? GetCourseById(int id)
+    public CourseDto? GetCourseById(int id)
     {
         var course = _courses.GetById(id);
-        return course;
+        return course != null ? new CourseDto(course) : null;
```

Comment:
It's easier for humans to read and understand positive conditions rather than negative ones. Lets use is null check instead of != null